client/src/pages/Search.js

```diff
@@ -51,547 +54,834 @@ const GQL_CREATE_SAVED_SEARCH = gql`
+      totalCount
+      list {
+        uuid
+        name
+        lat
+        lng
+      }
+    }
+  }
+`
 
-const SEARCH_CONFIG = {
-  [SEARCH_OBJECT_TYPES.REPORTS]: {
-    listName: `${SEARCH_OBJECT_TYPES.REPORTS}: reportList`,
-    listAllName: `all${SEARCH_OBJECT_TYPES.REPORTS}: reportList`,
-    sortBy: "ENGAGEMENT_DATE",
-    sortOrder: "DESC",
-    variableType: "ReportSearchQueryInput",
-    fields: ReportCollection.GQL_REPORT_FIELDS
-  },
-  [SEARCH_OBJECT_TYPES.PEOPLE]: {
-    listName: `${SEARCH_OBJECT_TYPES.PEOPLE}: personList`,
-    sortBy: "NAME",
-    sortOrder: "ASC",
-    variableType: "PersonSearchQueryInput",
-    fields:
-      "uuid, name, rank, role, emailAddress, avatar(size: 32), position { uuid, name, type, code, location { uuid, name }, organization { uuid, shortName} }"
-  },
-  [SEARCH_OBJECT_TYPES.POSITIONS]: {
-    listName: `${SEARCH_OBJECT_TYPES.POSITIONS}: positionList`,
-    sortBy: "NAME",
-    sortOrder: "ASC",
-    variableType: "PositionSearchQueryInput",
-    fields:
-      "uuid , name, code, type, status, location { uuid, name }, organization { uuid, shortName}, person { uuid, name, rank, role, avatar(size: 32) }"
-  },
-  [SEARCH_OBJECT_TYPES.TASKS]: {
-    listName: `${SEARCH_OBJECT_TYPES.TASKS}: taskList`,
-    sortBy: "NAME",
-    sortOrder: "ASC",
-    variableType: "TaskSearchQueryInput",
-    fields: "uuid, shortName, longName"
-  },
-  [SEARCH_OBJECT_TYPES.LOCATIONS]: {
-    listName: `${SEARCH_OBJECT_TYPES.LOCATIONS}: locationList`,
-    sortBy: "NAME",
-    sortOrder: "ASC",
-    variableType: "LocationSearchQueryInput",
-    fields: "uuid, name, lat, lng"
-  },
-  [SEARCH_OBJECT_TYPES.ORGANIZATIONS]: {
-    listName: `${SEARCH_OBJECT_TYPES.ORGANIZATIONS}: organizationList`,
-    sortBy: "NAME",
-    sortOrder: "ASC",
-    variableType: "OrganizationSearchQueryInput",
-    fields: "uuid, shortName, longName, identificationCode, type"
+const DEFAULT_PAGESIZE = 10
+
+const Pagination = props => {
+  const { pageNum, pageSize, totalCount, goToPage } = props
+  const numPages = pageSize <= 0 ? 1 : Math.ceil(totalCount / pageSize)
+  if (numPages === 1) {
+    return null
   }
+  return (
+    <header className="searchPagination">
+      <UltimatePagination
+        className="pull-right"
+        currentPage={pageNum + 1}
+        totalPages={numPages}
+        boundaryPagesRange={1}
+        siblingPagesRange={2}
+        hideEllipsis={false}
+        hidePreviousAndNextPageLinks={false}
+        hideFirstAndLastPageLinks
+        onChange={value => goToPage(value - 1)}
+      />
+    </header>
+  )
+}
+
+Pagination.propTypes = {
+  pageNum: PropTypes.number,
+  pageSize: PropTypes.number,
+  totalCount: PropTypes.number,
+  goToPage: PropTypes.func
 }
 
-class Search extends Page {
-  static propTypes = {
-    ...pagePropTypes,
-    setPagination: PropTypes.func.isRequired,
-    pagination: PropTypes.object
+const Organizations = props => {
+  const {
+    queryParams,
+    setTotalCount,
```

Comment:
I would give this property a different name, like handleSetTotalCount, or onSetTotalCount, just because now if I read setTotalCount , in combination with hooks, I would expect to have a useState hook for totalCount.

Reply:
Well, actually, it *does* call a state hook setter, it's just that the setter is provided outside the component.